app/assets/javascripts/directives/search_sidebar.js

```diff
@@ -0,0 +1,170 @@
+angular.module("OpenHq").directive("searchSidebar", function(Search) {
+  return {
+    restrict: "E",
+    template: JST['templates/directives/search_sidebar'],
+
+    controller: function($scope) {
+      $scope.searching = false; // currently performing a seach
+      $scope.term = ""; // what the user has searched for
+      $scope.count = 0; // number of results returned
+      $scope.morePages = false; // the search results include more pages
+      $scope.page = 0; // current page of results
+      $scope.loadingMore = false; // load more function is running
+
+      // Result templates
+      $scope.project_template = JST['templates/search/project'];
+      $scope.story_template = JST['templates/search/story'];
+      $scope.task_template = JST['templates/search/task'];
+      $scope.comment_template = JST['templates/search/comment'];
+      $scope.attachment_template = JST['templates/search/attachment'];
+
+      /**
+       * Mousetrap open/close the sidebar
+       */
+      Mousetrap.bind(['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'], function() {
+        $scope.openSearchSidebar();
+      });
+      Mousetrap.bind('esc', function() {
+        $scope.closeSearchSidebar();
+      });
+
+      /**
+       * Clicking outside the search sidebar closes it
+       */
+      $(document).on("click", function() {
+        $scope.closeSearchSidebar();
+      });
+      $(document).on("click", "search-sidebar", function(ev) {
+        ev.stopPropagation();
+      });
+
+      /**
+       * Pressing ESC while in the search sidebar input closes it
+       */
+      $(document).on('keyup', 'search-sidebar input', function(ev){
+        if (ev.keyCode == 27) $scope.closeSearchSidebar();
+      });
+
+      /**
+       * Clicking a link in the sidebar closes it
+       */
+      $(document).on('click', 'search-sidebar li a', function(){
```

Comment:
do this with ng-click if you can. May need to read the href off the link and do `$location.url = ''` on it if there isnt a way with angular to do ng-click but also follow the link